src/components/AddCompanies/AddCompanies.jsx

```diff
@@ -0,0 +1,93 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Button, Form, FormGroup, ControlLabel, FormControl, Col } from 'react-bootstrap';
+import './styles.less';
+
+const propTypes = {
+    addCompany: PropTypes.func.isRequired,
+};
+
+class AddCompanies extends Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			name: '',
+			address: '',
+			phone: '',
+			revenue: '',
+		};
+
+		this.addNewCompany = this.addNewCompany.bind(this);
+		this.handleFormControlChange = this.handleFormControlChange.bind(this);
+	}
+
+	addNewCompany(e) {
+		e.preventDefault();
+		const newCompany = {...this.state};
+		this.props.addCompany(newCompany)
+	}
+
+	handleFormControlChange(e) {
+		const { value, id } = e.target;
+		console.log('VALUE:', value);
+		console.log('FOR ITEM WITH NAME:', id);
+		this.setState({
+			[id]: value,
+		});
+	}
```

Comment:
Here I'm taking an event off of an input field, and destructuring the `value` and `id` properties off of the event's target. In doing so I can be clever about setting the state dynamically rather than setting up four functions with similar behavior. The `console.log` invocations can be removed, I left them there for you so you can see what's happening in the console when you type away at the input fields.